src/Symfony/Component/Messenger/Bridge/MongoDB/Transport/Sender.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\MongoDB\Transport;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Stamp\DelayStamp;
+use Symfony\Component\Messenger\Stamp\TransportMessageIdStamp;
+use Symfony\Component\Messenger\Transport\Sender\SenderInterface;
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+
+class Sender implements SenderInterface
+{
+    public function __construct(
+        private readonly Connection $connection,
+        private readonly SerializerInterface $serializer,
+    ) {
+    }
+
+    public function send(Envelope $envelope): Envelope
+    {
+        $encodedMessage = $this->serializer->encode($envelope);
+
+        /** @var DelayStamp|null $delayStamp */
+        $delayStamp = $envelope->last(DelayStamp::class);
+        $delay = null !== $delayStamp ? $delayStamp->getDelay() : 0;
+
+        if (!\array_key_exists('body', $encodedMessage)) {
+            throw new \InvalidArgumentException('Encoded message missing required "body" parameter');
+        }
+
+        if (!\is_string($encodedMessage['body'])) {
+            throw new \InvalidArgumentException('Encoded message must be a string, got '.\gettype($encodedMessage['body']));
```

Comment:
The exceptions from the Messenger namespace should be used.
```suggestion
            throw new \Symfony\Component\Messenger\Exception\InvalidArgumentException(\sprintf('Encoded message must be a string, got %s', get_debug_type($encodedMessage['body'])));
```